plotgraph.py

```diff
@@ -14,97 +14,148 @@
-                        lines = lines + [line]
-                        # Split the line into "words". 
-                        # http://stackoverflow.com/a/23720594/6771403
-                        # print("line = {0}".format(line))
-                        words_in_line = re.split("[, !?:;$#]+", line)
-                        # print("words = {0}".format(words_in_line))
-                        # Check if the word is a number. 
-                        # Write numbers to line dependend numbers variable.
-                        for word in words_in_line:
-                            if is_number(word):
-                                numbers_in_line = numbers_in_line + \
-                                                    [float(word)]
-                        # print("numbers_in_line = {0}".format(numbers_in_line))
-                        if numbers_in_line:
-                            # Take the i-th number in the line and put it into 
-                            # the i-th vector/list in vectors. 
-                            # If there is no i-th vector yet, create it before.
-                            for i in range(0,len(numbers_in_line),1):
-                                if not is_index(vectors, i): 
-                                    vectors.append([])
-                                vectors[i].append(numbers_in_line[i])
-                                # print("vectors = {0}".format(vectors))
-                if vectors:
-                    # Get setting for python executable.
-                    python_exec = self.settings().get('python_exec')
-                    # Get optional setting for library path.
-                    ld_library_path = self.settings().get('ld_library_path')
-                    if ld_library_path:
-                        python_exec = 'LD_LIBRARY_PATH={0} '.format(ld_library_path) + '"'+python_exec+'"'
-                        # print(python_exec)
-                    else:
-                        python_exec = '"'+python_exec+'"'
-                    window.run_command("exec", {"shell_cmd" : \
-                        '{0} "{1}/PlotGraph/plotvectors/plotvectors.py" -list_str="{2}"'.format(
-                            python_exec,
-                            sublime.packages_path(),
-                            vectors)})
-                    # Suppress the panel showing
-                    if self.settings().get("show_output_panel") == "False":
-                        window.run_command("hide_panel", {"panel": "output.exec"})
-        return None
\ No newline at end of file
+            if self.is_column_selection(view, selections):
+                selection_strings = [self.join_column_selections_to_string(view, selections)]
+            else:
+                selection_strings = (view.substr(selection) for selection in selections)
+
+            for selection_str in selection_strings:
+                vectors = []
+                if len(selection_str) > 4096:
```

Comment:
**Suggestion** We should avoid magic numbers. A bit of context for why this value was chosen could also be helpful.